Renderer/src/cz/fidentis/renderer/ComparisonGLEventListener.java

```diff
@@ -530,20 +535,25 @@ private void slicesAllRender() {
         gl.glClear(GL_DEPTH_BUFFER_BIT);
         if (info.isShowSamplingRays()) {
             gl.glColor4fv(info.getColorOfCut(), 0);
-            if (!info.getLists2().isEmpty()) {
-                for (LinkedList<Vector2f> l : info.getLists2().get(0)) {
-                    for (int i = 1; i < info.getSamplePoints().size(); i++) {
-                        if (info.getSamplePoints().get(i) != null) {
-                            gl.glLineWidth(info.getCutThickness() + 1);
-                            gl.glBegin(GL_LINES);
-                            gl.glVertex2d(info.getSamplePoints().get(i).x, info.getSamplePoints().get(i).y);
-                            gl.glVertex2d(l.get(i).x, l.get(i).y);
-                            gl.glEnd();
-                        }
-                    }
+            for (int i = 1; i < info.getSampleNormals().size(); i++) {
+                 if (info.getPointDistances()!= null && info.getPointDistances().get(i).size() > 1) {
+                    Vector2f n = new Vector2f(info.getSampleNormals().get(i));
+                    n.normalize();
+                    Vector2f a = new Vector2f(n);               
+                    a.scale(info.getPointDistances().get(i).get(0));
+                    a.add(info.getSamplePoints().get(i));
+                    Vector2f b = new Vector2f(n);
+                    b.scale(info.getPointDistances().get(i).get(info.getPointDistances().get(i).size() - 1));
+                    b.add(info.getSamplePoints().get(i));
+                    gl.glLineWidth(info.getCutThickness());
+                    gl.glBegin(GL_LINES);

```

Comment:
da sa glBegin/end vyhodit von z cyklu? dost to spomaluje program pokial sa to furt prepina